### DISCUSSION TITLE
Get enabled compiler flags

### DISCUSSION THREAD

**Discussion by mpreiner** [mesonbuild/meson#11403] [Q&A]
Is there a way to print a summary (or get a list) of the enabled compiler flags? I was looking at [compiler](https://mesonbuild.com/Reference-manual_returned_compiler.html), but I don't see any functionality to access the currently configured compile flags.

**Comment by andy5995** [comment]
I think what you want is in `builddir/compile_commands.json` after you've setup the builddir.

**Reply by mpreiner** [reply]
Yes, I'd like to get the flags that end up in `compile_commands.json`, but extract them to store it in a configure file. I was wondering whether there was a way via the meson DSL to access these flags.

**Comment by dcbaker** [comment]
Like, the ones from $CFLAGS or -Dc_args=? Get_option(‘c_args’) in the DSL will return an array of arguments

**Reply by mpreiner** [reply]
These are just the flags that are passed in from the user, but not what will be used to actually compile the libraries/executables, right?
